### DISCUSSION TITLE
Some Github pages deployment questions.

### DISCUSSION THREAD

**Discussion by kevindecapite** [squidfunk/mkdocs-material#2225] [Q&A] [answered]
I'm using the included GitHub Actions method of publishing my MkDocs to GitHub pages per the docs:

https://squidfunk.github.io/mkdocs-material/publishing-your-site/

My `.github/workflows/ci.yml` file has the following in it:

```
name: ci

on:
  push:
    branches:
      - master

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: 3.x
      - run: pip install mkdocs-material
      - run: mkdocs gh-deploy --force
```

This method of deployment is about 99% effective at the moment as there are 2 items that are not quite working:

1. For some reason my left sidebar navigation doesn't show on GH pages. I don't understand why it wouldn't show on GH pages while showing just fine in my local Docker version. Has anyone else seen this? Both GH and Docker are on the same commit.

2. I have my GitHub Pages settings for this repository configured to use a custom domain. However, every time I deploy, this value is removed from my GH Pages settings and I have to reconfigure it to make the site accessible again. What can I do about this behavior, if anything?

Otherwise I love using MkDocs and greatly appreciate its availability. Cheers.

**Comment by squidfunk** [comment, marked as answer]
1. Is the menu hamburger icon (top left) visible, i.e. does the sidebar collapse into the drawer which is also shown on mobile?) Did you maybe zoom into the page?
2. I believe you have to [create a `CNAME` file and add it to your repository](https://www.mkdocs.org/user-guide/deploying-your-docs/#custom-domains)

**Reply by kevindecapite** [reply]
Thank you for replying.

1. No there's no burger menu and I've checked the zoom level as I originally thought that may have been the case. Here's the live site: https://docs.lunaastrology.com/

I'm attaching a screen shot of my local Docker version.

2. I will move my `CNAME` file from the project root into `/docs` and see if that works 👍

**Reply by kevindecapite** [reply]
Is it also weird that even in the Docker version the main nav doesn't appear in the left sidebar?

![Screen Shot 2021-01-19 at 8 42 34 AM](https://user-images.githubusercontent.com/1189656/105065520-ce3e8300-5a32-11eb-80ae-8b352c1c66db.png)

**Reply by squidfunk** [reply]
I don't know exactly what you're aiming for, as the sidebar is clearly showing. Maybe you want to remove `navigation.tabs` from `mkdocs.yml`. Also, you may have different versions on GitHub and on your local installation. Ensure that the version of Material for MkDocs is identical.

**Reply by kevindecapite** [reply]
> the sidebar is clearly showing

Correct, but only on the Docker version. The live version doesn't show the sidebar:

https://docs.lunaastrology.com/

**Reply by squidfunk** [reply]
Ah sorry, I haven't checked the link as I thought the screenshot is the live version, sorry. That is not intentional. Could you please try to narrow it down to a minimum, reproducible test case and open an issue? Unfortunately, I don't have time to debug this in-depth, but if you manage to reproduce it with a minimal Material for MkDocs setup, I might be able to pin the error down and fix it quickly. I imagine it might be some combination of the navigation features (sections, tabs etc.).

**Reply by kevindecapite** [reply]
No apologies needed; I appreciate whatever attention you can give :-)

I'll post a reproducible test case as requested ASAP.

**Reply by kevindecapite** [reply]
Here's another deployment that's missing the left sidebar:

~~http://arcpointgroup.com/mkdocs-test/~~

It's a public repo so you can check it out locally and see how it operates for you. I'm seeing the same behavior on my end: Local Docker shows the left sidebar, GH Pages deployment does not.

~~https://github.com/arcpointgroup/mkdocs-test~~

[EDIT]: _Links are now dead._

**Reply by squidfunk** [reply]
Thanks for providing the repo. However, I'm afraid I currently don't have the time to do a build-deploy cycle to debug this 😅 Thus, if you manage to reproduce it _locally_ in a repository I can check out, `pip install` and `mkdocs serve`, I might be able to look into this.

**Reply by kevindecapite** [reply]
I understand, but of course it's only in the deployment that I'm seeing the problem. Locally everything works fine. 

If you can point me in the right direction for diagnosing this, that may be helpful.

**Reply by squidfunk** [reply]
Make sure that you have the same versions and try to reproduce it without Docker. It's likely some config edge case that wasn't considered. If we know what it is and how it reproduces locally without Docker, we can fix it.

**Reply by kevindecapite** [reply]
The problem is in the `navigation.tabs` feature. If I remove that, the left sidebar loads just fine. Here's a way to reproduce the issue (sans Docker):

__mkdocs.yml__

```
site_name: Test

theme:
  features:
    - navigation.tabs
  name: material
```

**Reply by squidfunk** [reply]
Thanks! I see now what you're aiming for. The tabs behavior was changed a few releases ago in order to allow nav entries and links to bee rendered as tabs. See https://github.com/squidfunk/mkdocs-material/issues/2125#issuecomment-744360768 for an explanation and possible solution.

The difference in behavior you got locally and in CI was likely due to different versions.

There's also a [footnote](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#fn:1) in the docs on this topic.

**Reply by kevindecapite** [reply]
Yes that's it. I was using an image from pre-December 2020. Once I updated it I'm now getting the same behavior in my Docker environment that I'm seeing in the production one. Of course, I see now this is the intended behavior so I'll just accommodate my docs implementation accordingly.

Thank you again for your attention and help with this.
